Build for arm64e and use older Xcode version when not packaging for rootless

thatmarcel opened the pull request:
For this tweak to work on A12 devices and up, `arm64e` needs to be added to the list of architectures.

It also seems like making this tweak work on some configurations requires changing the deployment target because, at least for me, the Settings app will crash when injecting a tweak with a deployment target of iOS 16, while a deployment target of iOS 15 works fine.

(On another note, when building with a deployment target of iOS 16 and `FINALPACKAGE=1`, the tweak does not seem to do anything at all, while iOS 15 and `FINALPACKAGE=1` works, so it might be worth looking into that?)

(I'm using a 2018 iPad Pro with iOS 14.5 for development)

The two above mentioned changes make the tweak run on my iPad, but it will still not work on A12+ devices running iOS 12.0-13.7 because of ABI changes. To support these devices, the tweak needs to be built with an Xcode version lower than 12 ([more here](https://theos.dev/docs/arm64e-deployment)).

I've added a condition to the Makefile that will use `/Applications/Xcode-11.7.app` when not packaging for rootless.

thatmarcel committed:
Build for arm64e and use older Xcode version when not packaging for rootless

thatmarcel commented:
To build the tweak using Xcode 11.7 via GitHub Actions, the [macOS 11 runner](https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md) could probably be used as it still ships with that version at `/Applications/Xcode_11.7.app`.

itsjunetime commented on Makefile:
```diff
@@ -1,7 +1,13 @@
+ARCHS = arm64 arm64e
+
+ifeq ($(THEOS_PACKAGE_SCHEME), rootless)
+	TARGET := iphone:clang:latest:16.0
```
```suggestion
	TARGET := iphone:clang:latest:15.0
```
Since you recommended using 15 instead of 16, I'll try that out in the CI of this branch to see if it seems to compile fine.

itsjunetime committed:
Change TARGET to 15.0 for rootless

itsjunetime committed:
Merge branch 'main' into rootful-older-xcode

itsjunetime committed:
Update to macos-11 runner to try to use older Xcode

itsjunetime commented:
looks like theos is trying to use the 16.5 sdk while we've told it to use a 15.x sdk, and that's causing some build issues. i ran into a similar issue at some point during development, but can't exactly remember what the fix was. i'll try to take a look at this later.

thatmarcel committed:
Fix GitHub actions build

Change Xcode 11.7 path for GitHub Actions, run rootless build on macOS 13 runner, use newer make version

thatmarcel commented:
I think we should probably still build the rootless tweak on a macOS 13 runner because we don't really have a reason not to. This means that the two builds will also run separately, which may be useful if only one of them fails. Also, for whatever reason, the Xcode 11.7 build fails if it's not using the newer `make` version installed by Homebrew, so I've added these changes to the workflow file, and it seems to build fine now.

thatmarcel committed:
Add note about THEOS_PACKAGE_SCHEME to readme

thatmarcel committed:
Add note about Xcode 11.7 to readme

thatmarcel commented:
Also added a note to the readme about setting `THEOS_PACKAGE_SCHEME=rootless` and about Xcode 11.7 to hopefully not confuse people trying to build the tweak.

itsjunetime commented:
Sorry for my silence for a bit, but this all looks great - thanks for adding this!
